File: app/src/main/java/io/railway/station/image/utils/StringUtils.java
package io.railway.station.image.utils;

public class StringUtils {

    // Получение сокращенного названия станции
    public static String getShortStation(String station, int limit) {
        if (station.length() <= limit) {
            return station;
        } else {
            String pattern = "аоеёиуэюя";
            int end = limit - 1;
            // Skip symbols from pattern
            while (pattern.indexOf(station.charAt(end)) != -1) {
                end--;
            }
            return station.substring(0, end + 1);
        }
    }

}

File: app/src/main/java/io/railway/station/image/utils/StringUtils.kt
package io.railway.station.image.utils


/**
 * Get short name
 */
fun String.limitWithLastConsonant(limit: Int) =
        if (length <= limit) this
        else {
            val pattern = "аоеёиуэюя"
            var index = limit - 1
            // Skip the vowels from pattern and find first consonant
            while (pattern.indexOf(get(index)) != -1) index--
            substring(0, index + 1)
        }


fun String?.ifNullOrEmpty(defValue: String) = if (isNullOrEmpty()) defValue else this